src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs

```diff
@@ -79,7 +79,7 @@ public static async ValueTask<Stream> ConnectAsync(string host, int port, Cancel
                 socket.NoDelay = true;
                 return new ExposedSocketNetworkStream(socket, ownsSocket: true);
             }
-            catch (Exception error)
+            catch (Exception error) when (!(error is OperationCanceledException))
```

Comment:
Nit: The logic here is essentially the same as the logic below, but it's implemented rather differently. "if" statement vs conditional expression, "when" clause vs "if" test.

I wonder if it's worthwhile to factor out this logic into a separate routine, so that it's clearly both correct and consistent. I suspect there are other places that are doing something similar when processing exceptions, and which may or may not be doing the right thing today.

Reply:
I could have done it with ifs in both places, but the if version is more code and isn't necessary here because there's no catch work to be done for the cancellation exceptions; in contrast below, there is work to be done for all failures.

Reply:
But, yeah, we can probably clean up / consolidate some of this code further.